realm/realm-annotations-processor/src/main/java/io/realm/processor/RealmProxyClassGenerator.java

```diff
@@ -414,8 +445,8 @@ private void emitValidateTableMethod(JavaWriter writer) throws IOException {
         writer.emitStatement("Table table = sharedRealm.getTable(\"%s%s\")", Constants.TABLE_PREFIX, this.simpleClassName);
 
         // verify number of columns
-        writer.beginControlFlow("if (table.getColumnCount() != " + metadata.getFields().size() + ")");
-        writer.emitStatement("throw new RealmMigrationNeededException(sharedRealm.getPath(), \"Field count does not match - expected %d but was \" + table.getColumnCount())",
+        writer.beginControlFlow("if (table.getColumnCount() < " + metadata.getFields().size() + ")");
+        writer.emitStatement("throw new RealmMigrationNeededException(sharedRealm.getPath(), \"Field count is less than expected - expected %d but was \" + table.getColumnCount())",
```

Comment:
Should we write to the log if count > getFields().size()? It is not critical but might be nice to have when debugging.